docs/noise.ipynb

```diff
@@ -62,196 +62,673 @@
+    "else:\n",
+    "    print(f\"Channel {channel} does not have a _mixture_ or _unitary_ method.\")"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {
-    "id": "ebed67376062"
+    "id": "6f3ae4337219"
    },
    "source": [
-    "### GeneralizedAmplitudeDampingChannel and AmplitudeDampingChannel\n",
-    "\n",
-    "The generalized amplitude damping channel models the effect of energy\n",
-    "dissipation to a surrounding environment as well as dephasing that\n",
-    "does not exchange energy. The amplitude damping channel only models dissipation of energy to a surrounding environment. \n",
+    "To summarize:\n",
+    "\n",
+    "* Every channel supports the `cirq.channel` protocol. (Note this includes gates like `cirq.X` and `cirq.CNOT`.)\n",
+    "  - If magic method `_channel_` is not defined, `cirq.channel` looks for `_mixture_` then for `_unitary_`.\n",
+    "* A subset of channels which support `cirq.channel` also support the `cirq.mixture` protocol.\n",
+    "  - If magic method `_mixture_` is not defined, `cirq.mixture` looks for `_unitary_`.\n",
+    "* A subset of channels which support `cirq.mixture` also support the `cirq.unitary` protocol.\n",
+    "  - If `_unitary_` is not defined for an object, the object is probably not meant for a circuit!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "0f5b77825a8a"
+   },
+   "source": [
+    "## Custom channels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "360d5a316769"
+   },
+   "source": [
+    "Channels not defined in `cirq.ops.common_channels` can be user-defined. Defining custom channels is similar to defining custom gates.\n",
     "\n",
-    "Cirq has implementations of both of these channels. The generalized amplitude damping channel corresponds to:\n",
+    "A minimal example for defining the channel\n",
     "\n",
     "$$\n",
-    "\\begin{aligned}\n",
-    "\\rho \\rightarrow& \\sum_{k=0}^3 M_k \\rho M_k \\newline\n",
-    "M_0 =& \\sqrt{p} \\begin{bmatrix} 1 & 0  \\cr 0 & \\sqrt{1 - \\gamma} \\end{bmatrix} \\newline\n",
-    "M_1 =& \\sqrt{p} \\begin{bmatrix} 0 & \\sqrt{\\gamma} \\cr 0 & 0 \\end{bmatrix} \\newline\n",
-    "M_2 =& \\sqrt{1-p} \\begin{bmatrix} \\sqrt{1-\\gamma} & 0 \\cr 0 & 1 \\\\ \\end{bmatrix} \\newline\n",
-    "M_3 =& \\sqrt{1-p} \\begin{bmatrix} 0 & 0 \\cr \\sqrt{\\gamma} & 0 \\end{bmatrix}\n",
-    "\\end{aligned}\n",
+    "\\rho \\mapsto (1 - p) \\rho + p Y \\rho Y\n",
     "$$\n",
     "\n",
-    "Where &#947; is the probability of the interaction being dissipative, and\n",
-    "`p` is the probability that the qubit and environment exchange energy. The amplitude damping channel corresponds to `p=1`.\n",
+    "is shown below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "55240879394e"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Minimal example of defining a custom channel.\"\"\"\n",
```

Comment:
Would it be possible to include examples of what one should provide if one wanted to define _unitary_(self) or _channel_(self)?

Reply:
Great point. Added a note on defining `_channel_` - let me know what you think. The case for defining `_unitary_` is documented in [custom gates](https://github.com/quantumlib/Cirq/blob/master/docs/custom_gates.ipynb).

Reply:
Looks good, thanks